examples/SubscriptionTest/main.cpp

```diff
@@ -33,13 +33,65 @@ int main()
         if (idx > 1) {
             cout << "Connected" << endl;
             UA_UInt32 subId = 0;
+
             if (client.addSubscription(subId)) {
                 cout << "Subscription Created id = " << subId << endl;
-                auto f = [](Open62541::ClientSubscription& c, Open62541::MonitoredItem *m, UA_DataValue* v) {
-                    cout << "Data Change SubId " << c.id()
-                         << " Monitor Item Id " << m->id()
-                         << " Value " << v->value.type->typeName << " "
-                         << Open62541::dataValueToString(v) << endl;
+                auto f = [&client,
+                          idx](Open62541::ClientSubscription& c, Open62541::MonitoredItem* m, UA_DataValue* v) {
+                    cout << "Data Change SubId " << c.id() << " Monitor Item Id " << m->id() << " Value "
+                         << v->value.type->typeName << " " << Open62541::dataValueToString(v) << endl;
+
+                    cout << endl;
+
+                    Open62541::NodeId nodeNumber(idx, "BOOLEAN_Value");
+                    Open62541::Variant nodeValue;
```

Comment:
I feel like the `Open62541::Variant` must be initialized with a type here!

Something like this. Apparently after reading the value, it does not know it has read a bool?!?!
```
Open62541::Variant<UA_TYPES_BOOLEAN>
```

Reply:
I also had my suspicions with this one, 
but the abstract Variant constructor does work with all types and initializes the internal data